ed448-goldilocks/src/edwards/affine.rs

```diff
@@ -100,6 +117,34 @@ impl AffinePoint {
         }
     }
 
+    /// Standard compression; store Y and sign of X
+    // XXX: This needs more docs and is `compress` the conventional function name? I think to_bytes/encode is?
+    pub fn compress(&self) -> CompressedEdwardsY {
+        let affine_x = self.x;
+        let affine_y = self.y;
+
+        let mut compressed_bytes = [0u8; 57];
+
+        let sign = affine_x.is_negative().unwrap_u8();
+
+        let y_bytes = affine_y.to_bytes();
+        compressed_bytes[..y_bytes.len()].copy_from_slice(&y_bytes[..]);
+        *compressed_bytes.last_mut().expect("at least one byte") = sign << 7;
+        CompressedEdwardsY(compressed_bytes)
+    }
+
+    /// Check if this point is on the curve
+    pub fn is_on_curve(&self) -> Choice {
+        // X^2 + Y^2 == 1 + D * X^2 * Y^2
+
+        let XX = self.x.square();
+        let YY = self.y.square();
+        let lhs = YY + XX;
+        let rhs = FieldElement::ONE + FieldElement::EDWARDS_D * XX * YY;
+
+        lhs.ct_eq(&rhs)
+    }
```

Comment:
Is this really needed, especially as a `pub fn`? The alternative approach used by the rest of the crates in this repo is to never instantiate an `AffinePoint` without first checking it's on the curve, and since Curve448/Ed448 always uses compressed coordinates for serialization, you effectively get that for free

Reply:
All the methods in this PR are just moved from `EdwardsPoint`, I didn't really touch them otherwise. My intention was to go over removing all these functions in a separate PR, tracking it in #1348.

But am happy to do it in this PR as well.